### Footstep parameterization for a large humanoid, similar to ASIMO or HRP-2 #
step_validation_config:
### footstep settings #########################################################
# Footstep displacement vectors (in meter / rad)
# WARNING: data type has to be float/double, integers will cause an exception


### step interpolation #############################################################
  interpolation:
    goal_position: {x: 1.2, y: 0.2, z: 0.0}


### foot settings #############################################################
  foot:
    size: {x: 0.2, y: 0.1, z: 0.05}
    separation: 0.3
    origin_shift: {x: 0.02, y: 0.0}
    max_step: {x: 0.22, y: 0.30, theta: 0.7, inverse_x: -0.1, inverse_y: 0.12, inverse_theta: -0.4}

  step_translation:
    max: {x: 0.22, y: 0.30, z: 0.2}
    min_inv: {x: -0.1, y: 0.0, z: -0.2}
  foot_rotation:
   max: 0.5
   inv: -0.19635
### step range ################################################################

# more accurate than the bounding box defined by the max/max inverse step
  step_range:
    x: [0.22, 0.22,-0.10,-0.10]
    y: [0.28, 0.12, 0.12, 0.28]

  angle:
    min: 0.0
    max: 6.28318452
    bins: 64


### step validation ################################################################

  validation:
    max_height_difference: 0.15
    max_incline: 0.35
    partial_foothold:
      min_foothold: 0.75
      height_deviation: 0.01
    step_collision_height: 0.01
    shin_collision_height: 0.07
    

  expansion:
    length: {x: 1.2, y: 0.5}
    position: {x: 0.6, y: 0.25}
    yaw: 0.0
